src/modules/Game.class.js

```diff
@@ -1,68 +1,265 @@
 'use strict';
 
-/**
- * This class represents the game.
- * Now it has a basic structure, that is needed for testing.
- * Feel free to add more props and methods if needed.
- */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  static Status = {
+    idle: 'idle',
+    playing: 'playing',
+    win: 'win',
+    lose: 'lose',
+  };
+
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.state = initialState.map((row) => [...row]);
+    this.status = Game.Status.idle;
+    this.initialState = initialState;
+    this.score = 0;
+  }
+
+  moveLeft() {
+    if (this.status === Game.Status.playing) {
+      let isMovable = false;
```

Comment:
Consider encapsulating the logic for merging and updating the rows into a separate method to avoid code duplication across `moveLeft`, `moveRight`, `moveUp`, and `moveDown` methods.